### DISCUSSION TITLE
Some questions that i had after reading and seeing how long it takes to build static engines.

### DISCUSSION THREAD

**Discussion by mr-pepe69** [comfyanonymous/ComfyUI_TensorRT#72] [Q&A] [answered]
i've seen this node and the huge performance it gives (when you manage to build the engines) is there a place where people share their engines? i've been having trouble building the engines. but they consume less vram that the normal model right? so isn't that possible to just share the .engine files?
this leads me to this questions:

Any chances .engine files will be able to be uploaded to civitai as their own filtered item? (like safetensor, picketensor and other)
how much ram building the engine uses compared to just running it? 
Does it use less ram than the same equivalent engine with the same settings?
Can an engine built in a 20 series run in a 30 series? (that comes from reading the read.me the wording sounded a bit confusing and i wanted to be sure)
i'll keep using it. or at least triying to. (my 3050 8gb suffers at lot with this kind of stuff)  and if i

and a bit as a joke.
is this going to be an hour long mistake?
![image](https://github.com/user-attachments/assets/60f013f6-3ad5-405d-afdd-21c0bab939e6)

**Comment by CloudWalker-II** [comment, marked as answer]
Unfortunately I can't answer most of your questions.
But as for sharing using .engine files others have built or between graphic cards; as I understand it, this is not possible.

You need to build the .engine file on the **exact** same graphic card that you then later use it on. As far as I know you can't even use it on another card from the same series _(so you can't even build on one 3050 card and then use it on another 3050 card)_.

Only other maybe helpful thing I can contribute with is that one my 10gb VRAM RTX 3080 it takes about 5 minutes _(give or take 5 ish seconds)_ to build a static SDXL .engine. From where I press the queue button to when it's finished.

From your image you have a lot going on, try building the engine in a separate workflow, with only the nodes you need to build it. Also that batch size seems really big, do you really need to generate 32 images every time? :P

Also try saving and then loading only the clip from the model to reduce the VRAM needed when using the RT .Engine as per this thread: https://github.com/comfyanonymous/ComfyUI_TensorRT/issues/15
